app/src/main/java/agency/five/codebase/android/movieapp/ui/favorites/FavoritesViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package agency.five.codebase.android.movieapp.ui.favorites
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.ui.favorites.mapper.FavoritesMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class FavoritesViewModel(
+    private val movieRepository: MovieRepository,
+    private val favoritesMapper: FavoritesMapper,
+) : ViewModel() {
+    val favoritesViewState: StateFlow<FavoritesViewState> =
+        movieRepository.favoriteMovies()
+            .map(favoritesMapper::toFavoritesViewState)
+            .stateIn(
+                scope = viewModelScope,
+                started = SharingStarted.WhileSubscribed(5_000),
+                initialValue = FavoritesViewState.EMPTY,
+            )
+
+    fun toggleFavorite(movieId: Int) {
+        viewModelScope.launch {
+            movieRepository.toggleFavorite(movieId)
```

Comment:
Since we can only remove (unlike) movies on this screen maybe there is more appropriate method you could use

Reply:
Does renaming the method to removeFavorite and using removeMovieFromFavorites make more sense in this case?